test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1200,19 +925,18 @@ private void setupExternalIdTest(boolean withValidation, boolean requiredOnSignU
     }
     
     private void verifyIdReservation(String withId) {
-        verify(externalIdService).reserveExternalId(STUDY, withId);
-        verify(externalIdService).assignExternalId(STUDY, withId, HEALTH_CODE);
-    }
-    
-    private void verifyIdAssignedWithoutReservation(String withId) {
-        verify(externalIdService, never()).reserveExternalId(STUDY, withId);
-        verify(externalIdService).assignExternalId(STUDY, withId, HEALTH_CODE);
-        verify(externalIdService, never()).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
+        verify(externalIdService).reserveExternalId(studyCaptor.capture(), eq(withId), eq(HEALTH_CODE));
+        verify(externalIdService).assignExternalId(studyCaptor.capture(), eq(withId), eq(HEALTH_CODE));
+        assertTrue(studyCaptor.getAllValues().get(0).isExternalIdValidationEnabled());
+        assertTrue(studyCaptor.getAllValues().get(1).isExternalIdValidationEnabled());
     }
     
-    private void verifyNotSetAsReservation() {
-        verify(externalIdService, never()).reserveExternalId(STUDY, EXTERNAL_ID);
-        verify(externalIdService, never()).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
+    private void verifyNotSetAsReservation(String externalId, boolean isUpdate) {
+        // This is now always called, so it's not so easy to verify.
+        if (!isUpdate) {
+            verify(externalIdService).reserveExternalId(STUDY, externalId, HEALTH_CODE);    
+        }
```

Comment:
Cleaned up tests (some of these helper methods aren't needed once you simplify).